activities/models/post_interaction.py

```diff
@@ -267,12 +341,40 @@ def by_ap(cls, data, create=False) -> "PostInteraction":
                 # Resolve the author
                 identity = Identity.by_actor_uri(data["actor"], create=True)
                 # Resolve the post
-                post = Post.by_object_uri(get_str_or_id(data["object"]), fetch=True)
+                object = data["object"]
+                target = get_str_or_id(object, "inReplyTo") or get_str_or_id(object)
+                post = Post.by_object_uri(target, fetch=True)
+                answer = None
                 # Get the right type
                 if data["type"].lower() == "like":
                     type = cls.Types.like
                 elif data["type"].lower() == "announce":
                     type = cls.Types.boost
+                elif (
+                    data["type"].lower() == "create"
+                    and object["type"].lower() == "note"
+                    and isinstance(post.type_data, QuestionData)
+                ):
+                    type = cls.Types.vote
+                    question = post.type_data
+                    answer = object["name"]
+                    if question.end_time and timezone.now() > question.end_time:
+                        # TODO: Maybe create an expecific expired exception?
+                        raise cls.DoesNotExist(
+                            f"Cannot create a vote to the expired question {post.id}"
+                        )
+
+                    already_voted = (
+                        post.type_data.mode == "oneOf"
+                        and post.interactions.filter(
+                            type=cls.Types.vote, identity=identity
+                        ).exists()
+                    )
+                    if already_voted:
```

Comment:
I believe this should change their vote, rather than raising an exception?

Reply:
I keep it like this to match the mastodon implementation that doesn't allow vote change once you voted, this is hard specially in the multiple choice case that sends choices in separate requests.

Reply:
Oh interesting, I thought Mastodon let you change the vote! We'll keep it this way then.